scalalib/src/mill/scalalib/JavaModule.scala

```diff
@@ -355,11 +355,11 @@ trait JavaModule
     }
 
   /**
-   * The output classfiles/resources from this module, excluding upstream
-   * modules and third-party dependencies
+   * The *output* classfiles/resources from this module, used for execution,
+   * excluding upstream modules and third-party dependencies
    */
   def localClasspath: T[Seq[PathRef]] = T {
-    compileResources() ++ resources() ++ Agg(compile().classes)
+    localCompileClasspath().toSeq ++ resources() ++ Agg(compile().classes)
```

Comment:
This change adds `unmanagedClasspath` to `localClasspath`. This is probably the right thing to do, since it means it'll probably be aggregated from upstream modules into `upstreamAssemblyClasspath`, where it wouldn't before